assignments/array-methods.js

```diff
@@ -53,31 +53,95 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 {"id":49,"first_name":"Bel","last_name":"Alway","email":"balway1c@ow.ly","shirt_size":"S","company_name":"Voolia","donation":107},
 {"id":50,"first_name":"Shell","last_name":"Baine","email":"sbaine1d@intel.com","shirt_size":"M","company_name":"Gabtype","donation":171}];
 
+// array1.forEach(function(element) {
+//     console.log(element);
+// });
+  
+
+console.log("Arrays");
+console.log("_________")
+// console.log("challenge 1");
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+// runners.forEach(cb);
+runners.forEach(function(ele) { // so this is a callback right?
+    fullName.push(`${ele.first_name} ${ele.last_name}`)
+});
+
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
+// allCaps = runners.map(cb); // cb should return something. eg return ele.first_name.toUpperCaser(); return whatever ur pushing
+runners.map(function(ele) {
+    allCaps.push(`${ele.first_name.toUpperCase()}`) // so wrong technically?
+});
+
+console.log(allCaps); // diff between .map and .forEach?
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+
+largeShirts = runners.filter(ele => ele.shirt_size === "L");
+
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
-console.log(ticketPriceTotal);
+let ticketPriceTotal = 0;
+
+const reducer = (accumulator, currentValue) => accumulator + currentValue.donation;
```

Comment:
Have you heard of reducers before? This type of function is indeed called a reducer. Come back to this code a month later and pat yourself on the back. You'll know why then.